schedview/compute/multisim.py

```diff
@@ -0,0 +1,366 @@
+        considererd "often visited".
+
+    Returns
+    -------
+    often_repeated_fields: `pandas.DataFrame`
+        The table of repeated field parameters, with the following columns:
+
+        ``fieldRA``
+            R.A. in degrees.
+        ``fieldDec``
+            Declination in degrees.
+        ``filter``
+            The filter.
+    often_repeated_field_stats: `pandas.DataFrame`
+        Statistics on the repetitions of each field.
+        The index is a ``pandas.MultiIndex`` with the following levels:
+
+        ``fieldRA``
+            R.A. in degrees.
+        ``fieldDec``
+            Declination in degrees.
+        ``filter``
+            The filter.
+        ``sim_index``
+            An ``int`` identifying which simulation a visit came from.
+
+        The columns are:
+
+        ``count``
+            The (``int``) number of visits in the simulation on the field.
+        ``first_time``
+            The ``datetime64[ns, UTC]`` start time of the first visit to the
+            field in the simulation.
+        ``last_time``
+            The ``datetime64[ns, UTC]`` start time of the last visit to the
+            field in the simulation.
+        ``label``
+            The ``str`` label for the simulation.
+    """
+    field_repeats: pd.DataFrame = visits.groupby(["fieldRA", "fieldDec", "filter", "sim_index"]).agg(
+        {"start_date": ["count", "min", "max"], "label": "first"}
+    )
+    column_map: dict[tuple[str, str], str] = {
+        ("start_date", "count"): "count",
+        ("start_date", "min"): "first_time",
+        ("start_date", "max"): "last_time",
+        ("label", "first"): "label",
+    }
+    # type hinting doesn't recognize that the column names in a multiindex
+    # are tuples.
+    field_repeats.columns = pd.Index([column_map[c] for c in field_repeats.columns])  # type: ignore
+
+    # Get the index in ra/dec/filter then use that as in index so we can show
+    # instances in simulations that have fewer than four visits of a field the
+    # is often visited in another simulation.
+    often_repeated_fields = (
+        field_repeats.query(f"count >= {min_counts}").droplevel("sim_index", "index").index.unique()
+    )
+
+    often_repeated_field_stats: pd.DataFrame = (
+        field_repeats.reset_index("sim_index")
+        .loc[often_repeated_fields, :]
+        .set_index("sim_index", append=True)
+    )
+    return often_repeated_fields.to_frame(), often_repeated_field_stats
+
+
+def count_visits_by_sim(
+    visits: pd.DataFrame,
+    sim_identifier_column: str = "sim_index",
+    visit_spec_columns: tuple[str, ...] = ("fieldRA", "fieldDec", "filter", "visitExposureTime"),
+) -> pd.DataFrame:
+    """Count the numbers of visits on each field in each simulation.
+
+    Parameters
+    ----------
+    visits : `pd.DataFrame`
+        A table that must include both the columns listed in
+        ``sim_identifier_column`` and ``visit_spec_columns`` (below).
```

Comment:
I'd typically call "visits" the visits from one simulation, so it may be worthwhile to add here  something like
"This data frame holds the visits read from multiple simulations, concatenated together."